cloud/pkg/devicecontroller/controller/upstream.go

```diff
@@ -226,10 +225,27 @@ func NewUpstreamController(dc *DownstreamController) (*UpstreamController, error
 	return uc, nil
 }
 
-func findTwinByName(twinName string, twins []v1beta1.Twin) *v1beta1.Twin {
-	for i := range twins {
-		if twinName == twins[i].PropertyName {
-			return &twins[i]
+func findOrCreateTwinByName(twinName string, Properties []v1beta1.DeviceProperty, DeviceStatus *DeviceStatus) *v1beta1.Twin {
+	for i := range Properties {
+		if twinName == Properties[i].Name {
+			twin := findTwinByName(twinName, DeviceStatus)
+			if twin != nil {
+				return twin
+			}
+			twin = &v1beta1.Twin{
+				PropertyName: twinName,
```

Comment:
I think of a scenario. If twin does not exist, a twin structure will be created here. If the propertyname defined in the yaml file of the device is test1 at the beginning, and later the user updates the propertyname to test2, will device.status still have test1 data at this time? Could you help test it?

Reply:
This consideration is very good. I have already made a judgment in the UpdateDevice function, and if twin cannot be found in properties, it will be removed.